### ISSUE TITLE
Analyzer 'Godot.SourceGenerators.MustBeVariantAnalyzer' threw an exception of type 'System.IndexOutOfRangeException'

### ISSUE THREAD

**Issue by tparviainen** [godotengine/godot#85553]
### Godot version

v4.2.stable.mono.official [46dc27791]

### System information

Ubuntu Linux

### Issue description

Source generator MustBeVariantAnalyzer throws exception System.IndexOutOfRangeException when generic attributes are declared in my own Godot project. ~~I think that generic attributes worked fine with Godot version 4.1.3.~~

### Steps to reproduce

1. Create new .NET Godot project
2. Add `<LangVersion>latest</LangVersion>` to csproj so that C# 'generic attributes' feature can be used
3. Create 2D scene with Node2D as the root node and rename this to for example `Level`
4. Attach script to `Level` node
5. Add next code to `Level.cs` file

```
[global::System.AttributeUsageAttribute(global::System.AttributeTargets.Class, Inherited = false, AllowMultiple = true)]
internal sealed class AutoloadAttribute<T> : global::System.Attribute
{
}
```

6. Decorate the `Level` class with the attribute i.e.

```
[Autoload<Level>]
public partial class Level : Node2D
{
}
```

7. Compile the project, output window has warning `Analyzer 'Godot.SourceGenerators.MustBeVariantAnalyzer' threw an exception of type 'System.IndexOutOfRangeException' with message 'Index was outside the bounds of the array.'`

If you compile the project from CLI using command `dotnet build --no-incremental -v d` you can see that the exception is thrown from method `MustBeVariantAnalyzer.ShouldCheckTypeArgument`.

### Minimal reproduction project

[GodotTest.zip](https://github.com/godotengine/godot/files/13515285/GodotTest.zip)

**Comment by AThousandShips** [comment]
> I think that generic attributes worked fine with Godot version 4.1.3.

What do you mean? Do you mean you recall that it maybe worked fine, or that your opinion is that it worked fine?

**Comment by tparviainen** [comment]
v4.1.3.stable.mono.official [f06b6836a] throws an exception as well.

**Comment by avilches** [comment]
> Thing is - none of the classes or methods in question are in the Godot context - the class isn't extending a godot type, and the attribute I'm defining is to do with my ECS library and not Godot. So it seems like the scope of what attributes should be analysed needs to be limited to just GodotSharp attributes?

The error is happening when it tries to extract the generic type before checking if the type is (or it isn't) annotated with the `[MustBeVariant]` attribute. Generic types in attributes belongs to the "ctor" method and the types are located in `ContainingType.TypeParameters` instead of `TypeParameters`. I've already solved it #85908
